### ISSUE TITLE
Ripping sprites and textures from Lego Mario/Luigi/Peach

### ISSUE THREAD

**Issue by ghost** [bricklife/LEGO-Mario-Reveng#6]
Not an issue, but yes a request: being able to extract sprites/textures that are only visible/usable on their chest, like flag animation among others.

I know the app has similar sprites, but are not equal and they aren't animated as they show up in the toy itself.

Thanks.

**Comment by jeffyjeffy1023** [comment]
If we can figure out some way to read the figure's firmware, it might be possible. I also wanna see this happen.

**Comment by jeffyjeffy1023** [comment]
There's probably some way to do it with a chip reader and a cracked-apart Luigi/Peach/Mario but that seems a bit impractical to me

**Comment by ghost** [comment]
I hope you or someone else make this possible.

**Comment by ghost** [comment]
@jeffyjeffy1023 Are you able to (if is the case) decrypt and extract files from lego mario and luigi .bin files?

I uploaded them, so, you can check them by yourself:

lego mario bin: https://cdn.discordapp.com/attachments/493153303551541258/963065479369478224/Mario_AssetPack-4-0-0-202.bin

lego luigi bin: https://cdn.discordapp.com/attachments/493153303551541258/963065570947919882/Luigi_AssetPack-4-0-0-215.bin

Thanks.

**Comment by jeffyjeffy1023** [comment]
Okay, firstly, how in the name of King Koopa did you get these?
Secondly, I've put the mario .bin into audacity as raw data and sometimes I can hear bits of Mario's voice.
Now, I'm thinking, if we can sorta get the audio by importing it as raw data, then we can probably also get the sprites that way too!
The problem is that I don't know how to interpret a file as an image because Google isn't telling me how to do so.
So for now I'll just keep playing around with the .bins and see what I can find, audio, sprites, or otherwise.
________________________________

**Comment by jeffyjeffy1023** [comment]
OK. I've somewhat managed to get clear audio from the mario bin.
The settings I used to import it into audacity this time was:

Encoding: Unsigned 8-bit PCM
Byte order: Default endianness
Channels: 1 Channel (Mono)
Start offset: 0 bytes
Amount to import: 100%
Sample rate: 44100 Hz

The audio is VERY sped up for some reason though, but I'll work on it.
________________________________

**Comment by jeffyjeffy1023** [comment]
Alright, so the way to play the audio correctly is this:
Music should be played at x0.18 speed
Voice clips should be played at around x0.36 speed
________________________________

**Comment by ghost** [comment]
Very well done on that! Nice news, indeed!

How can i export each track as .wav, for example? I'm new to that app!

I got those files from ios app, simple! lol

The next step is extract sprites from those bin files! =]

**Comment by Machow8** [comment]
Also, do you have a discord account? So, we can keep a more direct contact for all this. =]

**Comment by ghost** [comment]
This time i used watto's game extractor tool to extract files from bins, but they are still encrypted: https://www.mediafire.com/file/lgjoq7707y0uqyg/Lego_Mario_Luigi_files_%2528extracted_and_encrypted%2529.zip/file

But, this time will be better to just decrypt files instead of trying to find a way of dumping them.

**Comment by jeffyjeffy1023** [comment]
Opening up the files in audacity like I did before, it just kinda seems to me like Watto's game extractor just split the .bins into multiple files with little or no rhyme or reason, since it seems to me like it's just all the same, but scrambled around a bit.
________________________________

**Comment by ghost** [comment]
All the files are the same? Did you try to open all them in audacity to be sure?

**Comment by ghost** [comment]
Other good news: someone create a quickbms script which had let me extract files from those bins, this time being real files, not as you described.

This time, they are .dat files. I hope for the best that there is a tool which can extract their files or you can do that: https://cdn.discordapp.com/attachments/522932519524368404/964511022662680636/Lego_Mario_Luigi_files_dat.rar

**Comment by jeffyjeffy1023** [comment]
Yeah, opening the files in audacity with the same old settings again, it just seems like the quickbms script did the same thing watto's game extractor did.
Both don't actually​ know how to get proper files out of the bins.
the .bin extension means binary, meaning it's all raw data that only some internal tool at Lego/Nintendo would know how to properly work with.

Also, I'm sorry for using Audacity to actually test if there's any difference, (aside from the bin being cut up a bunch) I just don't know if there's any programs that can import a file as raw image data like Audacity can for audio, except maybe AviSynth's RawSource plugin, though I'm still trying to get that set up.
________________________________

**Comment by ghost** [comment]
Alright!

I thank you for helping me with all this!

Beyond Audacity, are you using another tool to try to extract sounds and textures from those files?

**Comment by ghost** [comment]
Should be better and more direct to just extract them from the toy, since they may be uncompressed in order to run on toy's simple hardware.

Try that.

**Comment by ghost** [comment]
Idk if this will help, but, i found a dummy bin from game files: https://cdn.discordapp.com/attachments/493153303551541258/964946774466899998/DummyAssetPack.bin

**Comment by jeffyjeffy1023** [comment]
I don't really think it will, since it's only 153 bytes and audacity ddoesn't give me anything than a short and somewhat faint "click" sound.
Though, if you can extract other stuff from the app, then try and extract everything you can from it I guess! Maybe in the app's files there's something that could help.
________________________________

**Comment by jeffyjeffy1023** [comment]
1.  Yeah, that setup I described is for audacity!
  2.  2. Yeah, I DO have a discord but I don't tend to talk on it that much, but it's jeffyjeffy1023#1568

________________________________

**Comment by BLiNXthetimesweeperGOD** [comment]
Late response, but these files aren't encrypted. The graphics are stored in 15 (or 16?) bits per pixel. I was able to get all of this with a tool known as tilemolester almost 2 years ago:
![LEGO Luigi progress](https://github.com/bricklife/LEGO-Mario-Reveng/assets/33909717/01332fe5-0a28-421e-a260-5318cc1c0d9e)

**Comment by BLiNXthetimesweeperGOD** [comment]
Oh, wait. That one had a ton of chunks deleted as I was assembling the sprites here:
![LEGO Luigi](https://github.com/bricklife/LEGO-Mario-Reveng/assets/33909717/2f085405-a81c-48c3-acee-aced89c0e329)
I might also start work on reversing the firmwares as well as the asset packages. The firmware also might've identified the hardware LEGO Mario runs on:
https://docs.zephyrproject.org/2.5.0/boards/arm/cc26x2r1_launchxl/doc/index.html
Oh, one more thing. I'm getting every single (known) firmware and asset package for Mario, Luigi and Peach uploaded to the Internet Archive:
https://archive.org/details/lego-mario-firmwares-and-asset-packs

**Comment by ghost** [comment]
@BLiNXthetimesweeperGOD

Wow! Amazing job! =D

Can you tell me stuff like software version, needed palette, steps to do/follow and which files you got those sprites from, exactly? =]

**Comment by BLiNXthetimesweeperGOD** [comment]
These don't use a palette, the mode is 2-Dimensional (makes data act more like variable width bitmaps than tilesets) and the codec is one of the 15bpp/16bpp ones. To get the right colors, you might need to go forward or back by a byte. Those sprites came from an asset package (all of the packages should have sprites and audio in them. I'm currently trying to figure those out too.) You'll have to mess with the width of the displayed graphics to get things to appear correctly.

**Comment by Machow8** [comment]
Do you have a discord.com account? I need some help on being able to properly replicate what you did.

Send your discord suername (username#1234) and i'll send a friendship request in there. =]

**Comment by ghost** [comment]
@BLiNXthetimesweeperGOD

Which tool did you use to extract those bin files from the base ones?

**Comment by BLiNXthetimesweeperGOD** [comment]
@Machow8 I forgot to reply, but this is what I use
https://github.com/toruzz/TileMolester

**Comment by ghost** [comment]
> @Machow8 I forgot to reply, but this is what I use https://github.com/toruzz/TileMolester

Alright!

So, what is all the setup you use in Tile Molester to properly rip those sprites with correct aspect ratio/size and color palette? =]

**Comment by JUDEPLAYS** [comment]
@BLiNXthetimesweeperGOD please upload Lego Marios and peaches
